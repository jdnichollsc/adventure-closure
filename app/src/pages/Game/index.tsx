import React from 'react';
import { IonContent, IonHeader, IonPage, IonTitle, IonToolbar } from '@ionic/react';

import { useFBInstant } from '../../hooks/useFBInstant';
import './style.css';
import GameContainer from '../../components/GameContainer';

const Game: React.FC = () => {
  const {
    initialized,
    started,
    error,
    progress,
    onStartGame,
    onLoadingProgress,
  } = useFBInstant()
  return (
    <IonPage>
      <IonHeader>
        <IonToolbar>
          <IonTitle>Adventure Closure</IonTitle>
        </IonToolbar>
      </IonHeader>
      <IonContent>
        <IonHeader collapse="condense">
          <IonToolbar>
            <IonTitle size="large">The game</IonTitle>
          </IonToolbar>
        </IonHeader>
        <GameContainer
          onStartGame={onStartGame}
          onLoadingProgress={onLoadingProgress}
        />
      </IonContent>
    </IonPage>
  );
};

export default Game;
