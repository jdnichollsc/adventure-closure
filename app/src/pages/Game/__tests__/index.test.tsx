import React from 'react'
import {
  mount,
  shallow
} from 'enzyme'

import Game from '..'

describe('Game page', () => {
  const defaultProps = {}

  it('should render', () => {
    const rendered = mount(<Game {...defaultProps} />)
    expect(rendered).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(shallow(<Game {...defaultProps} />)).toMatchSnapshot()
  })
})
