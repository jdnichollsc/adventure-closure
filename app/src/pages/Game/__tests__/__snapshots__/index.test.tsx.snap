// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game page renders correctly 1`] = `ShallowWrapper {}`;
